src/mapped-types.spec.snap.ts

```diff
@@ -479,14 +479,22 @@ type RequiredOptionalProps = {
     };
   };
   const nestedFunctionPartial: DeepPartial<NestedFunctionProps> = {};
-  // @dts-jest:pass:snap -> _DeepPartialObject<{ second: (value: number) => string; }> | undefined
+  // @dts-jest:pass:snap -> DeepPartial<{ second: (value: number) => string; }> | undefined
   testType<typeof nestedFunctionPartial.first>();
 
   const functionProp = nestedFunctionPartial.first!.second;
   // @dts-jest:pass:snap -> ((value: number) => string) | undefined
   testType<typeof functionProp>();
   // @dts-jest:pass:snap -> string
   testType<ReturnType<NonNullable<typeof functionProp>>>();
+
+  // @dts-jest:pass:snap -> DeepPartial<{ first: { second: { name: string; }; }; }>
+  testType<DeepPartial<NestedProps>>({});
+
+  <T extends object>() => {
+    // @dts-jest:pass:snap -> DeepPartial<{ first: { second: { name: string; }; }; }>
+    testType<DeepPartial<NestedProps>>({});
+  };
```

Comment:
@mpiniarski I think this test should be like this to reflect the original issue report, could you try it?

```ts
<T extends NestedProps>() => {
    // @dts-jest:pass:snap -> I'm not sure about what type will be resulting here from dts, possibly ` DeepPartial<T>`, let me know...
    testType<DeepPartial<T>>({});
  };
```